### DISCUSSION TITLE
Can't create trigger for table in auth schema

### DISCUSSION THREAD

**Discussion by bradparker** [supabase/supabase#18159] [Questions]
I've been following along with this course https://egghead.io/courses/build-a-twitter-clone-with-the-next-js-app-router-and-supabase-19bebadb, to try and learn a thing or two about Supabase. I hit a snag it seems a few other people have hit: when trying to create a trigger through the UI only tables in the public schema are available as options.

I took the advice offered in some threads online about this issue and tried to create the trigger via the SQL editor but it only reports "Error running SQL: Invalid SQL query"

Here's the full content of the function being triggered, I created this via the UI:

```sql
CREATE OR REPLACE FUNCTION public.insert_profile_for_new_user()
  RETURNS trigger
  LANGUAGE plpgsql
  SECURITY DEFINER
  AS $function$begin
    insert into public.profiles(id, name, username, avatar_url)
    values(
      new.id,
      new.raw_user_meta_data->>'name',
      new.raw_user_meta_data->>'user_name',
      new.raw_user_meta_data->>'avatar_url'
    );
    return new;
end;$function$
```

And here's the trigger I'm trying to create (running this in the SQL editor just reports: "Error running SQL: Invalid SQL query"):

```sql
create or replace trigger on_auth.users_insert
  after insert on auth.users
  for each row
  execute function public.insert_profile_for_new_user();
```

Any and all help greatly appreciated.

**Comment by GaryAustin1** [comment]
You should not use . in the name of the trigger.  It is confusing it as it thinks a schema is involved or some such.  Just give it a name with a period or put it in double quotes like "on_auth.users_insert"
